### ISSUE TITLE
#nullable enable should detect nested initialization of fields

### ISSUE THREAD

**Issue by mengluwang121** [dotnet/roslyn#35592]
**Version Used**: 
C# 8.0, .Net Core SDK 2.1.700

**Steps to Reproduce**:
With #nullable enable, "Non-nullable field is uninitialized" warning will show up if some field is not initialized in constructor. But sometimes we want to initialize some thing in a function so that it can be well structured, or we need to call an abstract function on children class. So this is bug is filed as a suggestion to consider detect nested initialization for #nullable enable. Thank you so much!
Code example: 

```
 public PipeProgressEventArgs(Stream stream)​
            : base(stream)​
        {​
            Initialize();​
        }​
​
        private void Initialize()​
        {​
            Progress = new ProgressEventArgs("Package", 12, ProgressType.Install);​
        }​
```

**Expected Behavior**:
No warning.
**Actual Behavior**:
 "Non-nullable field is uninitialized" shows up around the constructor.

**Comment by jcouv** [comment]
@mengluwang121 Yes, we recognize there is a pain point around this warning as many initialization patterns exist and the compiler currently produces warnings when it fails to understand them. We're looking to refine this.

**Comment by jaredpar** [comment]
Dupe of #32358
